tests/models/whisper/test_modeling_whisper.py

```diff
@@ -2136,18 +2208,76 @@ def test_whisper_longform_multi_batch(self):
+        # fmt: on
+
+        processor = WhisperProcessor.from_pretrained("openai/whisper-tiny")
+        model = WhisperForConditionalGeneration.from_pretrained("openai/whisper-tiny")
+        model = model.to("cuda")
+
+        ds = load_dataset("patrickvonplaten/librispeech_asr_dummy", "clean")
+        one_audio = np.concatenate([x["array"] for x in ds["validation"]["audio"]], dtype=np.float32)
+        audios = []
+        audios.append(one_audio[110000:])
+        audios.append(one_audio[:800000])
+        audios.append(one_audio[80000:])
+        audios.append(one_audio[:])
+
+        gen_kwargs = {"return_timestamps": True}
+        gen_kwargs["no_speech_threshold"] = 0.6
+        gen_kwargs["temperature"] = 0.0
+        gen_kwargs["compression_ratio_threshold"] = 1.35
+        gen_kwargs["condition_on_prev_tokens"] = True
+        gen_kwargs["logprob_threshold"] = -1.0
+
+        with open("/home/patrick/expected.txt", "w") as f:
+            decoded_single = []
+            for audio in audios:
+                inputs = processor(audio, return_tensors="pt", truncation=False)
+                inputs = inputs.to(device="cuda")
+
+                result = model.generate(**inputs, **gen_kwargs)
+                decoded_single.append(processor.batch_decode(result, skip_special_tokens=True))
+                f.write(decoded_single[-1][0] + "\n")
```

Comment:
Why do we need to write the output list here?

Reply:
ups nice catch! Will remove